framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,433 @@
+package org.checkerframework.common.value;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+
+/**
+ * The Range Class with mathematics operations. Models the range indicated by the @IntRange
+ * annotation.
+ *
+ * <p>Note that this class is not responsible for detecting incorrect parameters, e.g. the value of
+ * "from" could be greater than the value of "to". This incorrectness would be eventually caught by
+ * {@link
+ * org.checkerframework.common.value.ValueAnnotatedTypeFactory#createIntRangeAnnotation(Range)} when
+ * creating an annotation from range, which would then be replaced with @UnknownVal.
+ *
+ * @author JasonMrX
+ */
+public class Range {
+
+    /** The value 'from' */
+    public final long from;
+
+    /** The value 'to' */
+    public final long to;
+
+    /**
+     * Constructs a range with its bounds specified by two parameters, "from" and "to".
+     *
+     * <p>Note that it is possible to construct a range with incorrect parameters, e.g. the value of
+     * "from" could be greater than the value to "to". This incorrectness would be caught by {@link
+     * org.checkerframework.common.value.ValueAnnotatedTypeFactory#createIntRangeAnnotation(Range)}
+     * when creating an annotation from range, which would then be replaced with @UnknownVal.
+     *
+     * @param from the lower bound (inclusive)
+     * @param to the higher bound (inclusive)
+     */
+    public Range(long from, long to) {
+        this.from = from;
+        this.to = to;
+    }
+
+    public Range(long value) {
```

Comment:
The Range constructors with zero and one argument are never used.  I would therefore omit them from the code, to keep it simple and understandable.